MPChartLib/src/com/github/mikephil/charting/data/ChartData.java

```diff
@@ -95,41 +97,41 @@ public ChartData(ArrayList<String> xVals, ArrayList<T> sets) {
     }
 
     /**
-     * constructor that takes string array instead of arraylist string
+     * constructor that takes string array instead of List string
      * 
      * @param xVals The values describing the x-axis. Must be at least as long
      *            as the highest xIndex in the Entry objects across all
      *            DataSets.
      * @param sets the dataset array
      */
-    public ChartData(String[] xVals, ArrayList<T> sets) {
-        this.mXVals = arrayToArrayList(xVals);
+    public ChartData(String[] xVals, List<T> sets) {
+        this.mXVals = arrayToList(xVals);
         this.mDataSets = sets;
 
         init(mDataSets);
     }
 
     /**
-     * Turns an array of strings into an arraylist of strings.
+     * Turns an array of strings into an List of strings.
      * 
      * @param array
      * @return
      */
-    private ArrayList<String> arrayToArrayList(String[] array) {
+    private List<String> arrayToList(String[] array) {
 
```

Comment:
Yeah, that'd make much more sense. I didn't go into analyzing improvements or if the code makes sense. Thanks!